pkg/solana/chainwriter/lookups.go

```diff
@@ -98,7 +98,17 @@ type AccountsFromLookupTable struct {
 	LookupOpts
 }
 
-func (ac AccountConstant) Resolve(_ context.Context, _ any, _ map[string]map[string][]*solana.AccountMeta, _ client.Reader) ([]*solana.AccountMeta, error) {
+type ATALookup struct {
+	// Field that determines whether the ATA lookup is necessary. Basically
+	// just need to check this field exists. Dot separated location.
+	Location string
+	// If the field exists, initialize a ATA account using the Wallet, Token Program, and Mint addresses below
```

Comment:
I'm getting confused by these 3 fields.
Ideally, when we do a Token Transfer, aren't these fields already derivable from the CCIP message, right?

For example Any2SVMRampMessage.TokenReceiver, and Any2SVMTokenTransfer.DestTokenAddress are the fields.
I think 
WalletAddress = Any2SVMRampMessage.TokenReceiver
TokenProgram = Any2SVMTokenTransfer.DestTokenAddress
MintAddress = ?

So is it right to assume that these 3 fields you added will correspond to the location in the message, for these fields?

Another thing to note is that in CCIP, the TokenReceiver is just 1 entity, but multiple TokenPrograms.

Reply:
I think there is a bit of confusion on what these fields are.
1. WalletAddress above is correct.
2. TokenProgram isn't the DestTokenAddress, rather it is the program the token program that the dest token implements. I believe on solana this can either be the pre-2022 Token Program or the Token 2022 program. There may be more added in the future. I believe this is analagous to the ERC20, ERC667, etc. interfaces on EVM. Therefore we have to look this value up with the token address.
3. The MintAddress is the token address itself, i.e. the DestTokenAddress

Reply:
> Another thing to note is that in CCIP, the TokenReceiver is just 1 entity, but multiple TokenPrograms.

Yes this is a valid and expected case. As you can see in the code [here](https://github.com/smartcontractkit/chainlink-solana/blob/2df78b7c9b27939849643f9555136e7d8639780e/pkg/solana/chainwriter/chain_writer.go#L253-L266) we check that there is a single wallet yet there can be multiple tokenPrograms and mints

Reply:
So if TokenProgram can be looked up from the wallet address, then is it possible to remove from here?

Reply:
I don't quite understand what you're suggesting. How would we define the lookup to TokenProgram if we remove it here?